clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java

```diff
@@ -4346,6 +4346,51 @@ public void testListOffsetsUnsupportedNonMaxTimestamp() {
         }
     }
 
+    @Test
+    public void testListOffsetsNonMaxTimestampDowngradedImmediately() throws Exception {
+        Node node = new Node(0, "localhost", 8120);
+        List<Node> nodes = Collections.singletonList(node);
+        List<PartitionInfo> pInfos = new ArrayList<>();
+        pInfos.add(new PartitionInfo("foo", 0, node, new Node[]{node}, new Node[]{node}));
+        final Cluster cluster = new Cluster(
+                "mockClusterId",
+                nodes,
+                pInfos,
+                Collections.emptySet(),
+                Collections.emptySet(),
+                node);
+        final TopicPartition tp0 = new TopicPartition("foo", 0);
+
+        try (AdminClientUnitTestEnv env = new AdminClientUnitTestEnv(cluster,
+                AdminClientConfig.RETRIES_CONFIG, "2")) {
+
+            ApiVersion oldBrokerVersion = new ApiVersion();
+            oldBrokerVersion.setApiKey(ApiKeys.LIST_OFFSETS.id);
+            oldBrokerVersion.setMinVersion((short) 0);
+            oldBrokerVersion.setMaxVersion((short) 6);
```

Comment:
Using min/max versions is pretty neat. I wonder if we should update the other tests to use this as well. It feels a bit more explicit. What do you think?

Reply:
agreed, I've updated them